libwasmvm/src/storage.rs

```diff
@@ -65,7 +65,7 @@ impl Storage for GoStorage {
         order: Order,
     ) -> BackendResult<u32> {
         let mut error_msg = UnmanagedVector::default();
-        let mut iter = GoIter::new(self.db.gas_meter);
+        let mut iter = GoIter::stub();
```

Comment:
Setting the gas meter here to an otherwise unusable instance was inconsistent and confusing. Since `scan_db` actually created the iterator in Go, we can also build the full `C.GoIter` there.

Reply:
Instead of a stub, we could also use `MaybeUninit<GoIter>`, but it doesn't really matter much in this case.

Reply:
I had a brief look into it but never worked with this API. It sounded a bit scary to me, also when it comes to drop behaviour. But happy to adjust if there is a benefit.

Reply:
The benefit is very miniscule. It just forces you to be more explicit about when you consider the value initialized, since you have to explicitly call `assume_init` to get to the contained value. But that's not a huge problem here, since it's pretty clear where we pass the mutable pointer, so let's keep it like it is.
The drop behaviour is a good point: `MaybeUninit` will not drop. So, dropping only occurs after the `assume_init` call "unpacks" the value. A panic in-between writing the value and calling `assume_init` could lead to a leak. In our case the `GoIter` lives on the stack, so this is not a problem.